### ISSUE TITLE
MD033 Inline HTML false positive

### ISSUE THREAD

**Issue by Owen-OptiGrid** [rvben/rumdl#90]
HTML tags (or apparent tags) inside code blocks normally don't trigger MD033 (as expected) - but I found there's some weird interaction when multiple code blocks are inside a markdown file such that MD033 will start incorrectly flagging inline HTML.

Here's a minimal example file I managed to reproduce the issue with using rumdl 0.0.146:

````markdown
# Heading

`<env>`

```diff
- this
+ that
```
````

Note without the diff code block this will pass fine. With the diff block I get an error:

```shell
rumdl check test.md
test.md:3:2: [MD033] Inline HTML found: <env> (use Markdown syntax instead)

Issues: Found 1 issues in 1 file (9ms)
```

**Comment by rvben** [comment]
Hi @Owen-OptiGrid,

Thanks for the issue.

The problem was a 1-indexed vs 0-indexed confusion. This was a clear bug with a clear fix. Added tests to prevent this in the future.

Thanks for making rumdl better ! :)
